Fail gracefully if discord natives cannot load

Some architectures Discord does not release binaries for, like aarch64 , which causes the client to fail to start.

```1) Error injecting constructor, java.lang.UnsatisfiedLinkError: Unable to load library 'discord-rpc':```

`potato@odroid64:~$ wget https://github.com/runelite/launcher/releases/download/1.4.2/RuneLite.jar && java -jar RuneLite.jar --debug
--2018-03-04 21:19:24--  https://github.com/runelite/launcher/releases/download/1.4.2/RuneLite.jar
Resolving github.com (github.com)... 192.30.253.113, 192.30.253.112
Connecting to github.com (github.com)|192.30.253.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/57470030/c39a7650-18ce-11e8-8851-0cc0016ed459?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20180305%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180305T031924Z&X-Amz-Expires=300&X-Amz-Signature=8de9f664dd10a0777c62b984764e46a76800abbad98eba3c44177f49ab2e94ad&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3DRuneLite.jar&response-content-type=application%2Foctet-stream [following]
--2018-03-04 21:19:24--  https://github-production-release-asset-2e65be.s3.amazonaws.com/57470030/c39a7650-18ce-11e8-8851-0cc0016ed459?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20180305%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20180305T031924Z&X-Amz-Expires=300&X-Amz-Signature=8de9f664dd10a0777c62b984764e46a76800abbad98eba3c44177f49ab2e94ad&X-Amz-SignedHeaders=host&actor_id=0&response-content-disposition=attachment%3B%20filename%3DRuneLite.jar&response-content-type=application%2Foctet-stream
Resolving github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)... 52.216.230.107
Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (github-production-release-asset-2e65be.s3.amazonaws.com)|52.216.230.107|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2163494 (2.1M) [application/octet-stream]
Saving to: ‘RuneLite.jar.3’

RuneLite.jar.3      100%[===================>]   2.06M  1.92MB/s    in 1.1s    

2018-03-04 21:19:26 (1.92 MB/s) - ‘RuneLite.jar.3’ saved [2163494/2163494]

[main] INFO net.runelite.launcher.Launcher - Verified signature of net.runelite:client:jar:1.2.18
[main] INFO net.runelite.launcher.JvmLauncher - Using Java version 1.x
[main] INFO net.runelite.launcher.JvmLauncher - Running [/usr/lib/jvm/java-8-openjdk-arm64/jre/bin/java, -cp, /home/potato/.runelite/repository/net/runelite/client/1.2.18/client-1.2.18.jar:/home/potato/.runelite/repository/org/slf4j/slf4j-api/1.7.12/slf4j-api-1.7.12.jar:/home/potato/.runelite/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar:/home/potato/.runelite/repository/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar:/home/potato/.runelite/repository/net/sf/jopt-simple/jopt-simple/5.0.1/jopt-simple-5.0.1.jar:/home/potato/.runelite/repository/com/google/guava/guava/23.2-jre/guava-23.2-jre.jar:/home/potato/.runelite/repository/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.jar:/home/potato/.runelite/repository/com/google/errorprone/error_prone_annotations/2.0.18/error_prone_annotations-2.0.18.jar:/home/potato/.runelite/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/potato/.runelite/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/potato/.runelite/repository/com/google/inject/guice/4.1.0/guice-4.1.0-no_aop.jar:/home/potato/.runelite/repository/javax/inject/javax.inject/1/javax.inject-1.jar:/home/potato/.runelite/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar:/home/potato/.runelite/repository/com/google/code/gson/gson/2.4/gson-2.4.jar:/home/potato/.runelite/repository/org/pushingpixels/substance/8.0.00-dev/substance-8.0.00-dev.jar:/home/potato/.runelite/repository/org/pushingpixels/trident/1.4/trident-1.4.jar:/home/potato/.runelite/repository/net/runelite/api/1.2.18/api-1.2.18.jar:/home/potato/.runelite/repository/net/runelite/rs/api/1.2.18/api-1.2.18.jar:/home/potato/.runelite/repository/net/runelite/injected-client/1.2.18/injected-client-1.2.18.jar:/home/potato/.runelite/repository/net/runelite/scripts/1.2.18/scripts-1.2.18.jar:/home/potato/.runelite/repository/net/runelite/http-api/1.2.18/http-api-1.2.18.jar:/home/potato/.runelite/repository/com/squareup/okhttp3/okhttp/3.7.0/okhttp-3.7.0.jar:/home/potato/.runelite/repository/com/squareup/okio/okio/1.12.0/okio-1.12.0.jar:/home/potato/.runelite/repository/net/runelite/discord/1.0/discord-1.0.jar:/home/potato/.runelite/repository/net/java/dev/jna/jna/4.5.1/jna-4.5.1.jar, -Xmx512m, -Xss2m, -Dsun.java2d.noddraw=false, -XX:CompileThreshold=1500, -Xincgc, -XX:+UseConcMarkSweepGC, -XX:+UseParNewGC, -Djna.nosys=true, net.runelite.client.RuneLite]
OpenJDK 64-Bit Server VM warning: Using incremental CMS is deprecated and will likely be removed in a future release
Exception in thread "main" com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.UnsatisfiedLinkError: Unable to load library 'discord-rpc': Native library (linux-aarch64/libdiscord-rpc.so) not found in resource path ([file:/home/potato/.runelite/repository/net/runelite/client/1.2.18/client-1.2.18.jar, file:/home/potato/.runelite/repository/org/slf4j/slf4j-api/1.7.12/slf4j-api-1.7.12.jar, file:/home/potato/.runelite/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar, file:/home/potato/.runelite/repository/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar, file:/home/potato/.runelite/repository/net/sf/jopt-simple/jopt-simple/5.0.1/jopt-simple-5.0.1.jar, file:/home/potato/.runelite/repository/com/google/guava/guava/23.2-jre/guava-23.2-jre.jar, file:/home/potato/.runelite/repository/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.jar, file:/home/potato/.runelite/repository/com/google/errorprone/error_prone_annotations/2.0.18/error_prone_annotations-2.0.18.jar, file:/home/potato/.runelite/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar, file:/home/potato/.runelite/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar, file:/home/potato/.runelite/repository/com/google/inject/guice/4.1.0/guice-4.1.0-no_aop.jar, file:/home/potato/.runelite/repository/javax/inject/javax.inject/1/javax.inject-1.jar, file:/home/potato/.runelite/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar, file:/home/potato/.runelite/repository/com/google/code/gson/gson/2.4/gson-2.4.jar, file:/home/potato/.runelite/repository/org/pushingpixels/substance/8.0.00-dev/substance-8.0.00-dev.jar, file:/home/potato/.runelite/repository/org/pushingpixels/trident/1.4/trident-1.4.jar, file:/home/potato/.runelite/repository/net/runelite/api/1.2.18/api-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/rs/api/1.2.18/api-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/injected-client/1.2.18/injected-client-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/scripts/1.2.18/scripts-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/http-api/1.2.18/http-api-1.2.18.jar, file:/home/potato/.runelite/repository/com/squareup/okhttp3/okhttp/3.7.0/okhttp-3.7.0.jar, file:/home/potato/.runelite/repository/com/squareup/okio/okio/1.12.0/okio-1.12.0.jar, file:/home/potato/.runelite/repository/net/runelite/discord/1.0/discord-1.0.jar, file:/home/potato/.runelite/repository/net/java/dev/jna/jna/4.5.1/jna-4.5.1.jar])
  at net.runelite.client.discord.DiscordService.<init>(Unknown Source)
  at net.runelite.client.discord.DiscordService.class(Unknown Source)
  while locating net.runelite.client.discord.DiscordService
    for field at net.runelite.client.RuneLite.discordService(Unknown Source)
  at net.runelite.client.RuneLite.class(Unknown Source)
  while locating net.runelite.client.RuneLite

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1028)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1054)
	at net.runelite.client.RuneLite.main(RuneLite.java:123)
Caused by: java.lang.UnsatisfiedLinkError: Unable to load library 'discord-rpc': Native library (linux-aarch64/libdiscord-rpc.so) not found in resource path ([file:/home/potato/.runelite/repository/net/runelite/client/1.2.18/client-1.2.18.jar, file:/home/potato/.runelite/repository/org/slf4j/slf4j-api/1.7.12/slf4j-api-1.7.12.jar, file:/home/potato/.runelite/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar, file:/home/potato/.runelite/repository/ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar, file:/home/potato/.runelite/repository/net/sf/jopt-simple/jopt-simple/5.0.1/jopt-simple-5.0.1.jar, file:/home/potato/.runelite/repository/com/google/guava/guava/23.2-jre/guava-23.2-jre.jar, file:/home/potato/.runelite/repository/com/google/code/findbugs/jsr305/1.3.9/jsr305-1.3.9.jar, file:/home/potato/.runelite/repository/com/google/errorprone/error_prone_annotations/2.0.18/error_prone_annotations-2.0.18.jar, file:/home/potato/.runelite/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar, file:/home/potato/.runelite/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar, file:/home/potato/.runelite/repository/com/google/inject/guice/4.1.0/guice-4.1.0-no_aop.jar, file:/home/potato/.runelite/repository/javax/inject/javax.inject/1/javax.inject-1.jar, file:/home/potato/.runelite/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar, file:/home/potato/.runelite/repository/com/google/code/gson/gson/2.4/gson-2.4.jar, file:/home/potato/.runelite/repository/org/pushingpixels/substance/8.0.00-dev/substance-8.0.00-dev.jar, file:/home/potato/.runelite/repository/org/pushingpixels/trident/1.4/trident-1.4.jar, file:/home/potato/.runelite/repository/net/runelite/api/1.2.18/api-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/rs/api/1.2.18/api-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/injected-client/1.2.18/injected-client-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/scripts/1.2.18/scripts-1.2.18.jar, file:/home/potato/.runelite/repository/net/runelite/http-api/1.2.18/http-api-1.2.18.jar, file:/home/potato/.runelite/repository/com/squareup/okhttp3/okhttp/3.7.0/okhttp-3.7.0.jar, file:/home/potato/.runelite/repository/com/squareup/okio/okio/1.12.0/okio-1.12.0.jar, file:/home/potato/.runelite/repository/net/runelite/discord/1.0/discord-1.0.jar, file:/home/potato/.runelite/repository/net/java/dev/jna/jna/4.5.1/jna-4.5.1.jar])
	at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:303)
	at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:427)
	at com.sun.jna.Library$Handler.<init>(Library.java:179)
	at com.sun.jna.Native.loadLibrary(Native.java:569)
	at com.sun.jna.Native.loadLibrary(Native.java:544)
	at net.runelite.discord.DiscordRPC.<clinit>(DiscordRPC.java:42)
	at net.runelite.client.discord.DiscordService.<init>(DiscordService.java:59)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.google.inject.internal.DefaultConstructionProxyFactory$ReflectiveProxy.newInstance(DefaultConstructionProxyFactory.java:126)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:111)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:268)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:120)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:90)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:268)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1019)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1015)
	... 2 more
`

--- a/runelite-client/src/main/java/net/runelite/client/discord/DiscordService.java
+++ b/runelite-client/src/main/java/net/runelite/client/discord/DiscordService.java
@@ -55,8 +55,7 @@ public class DiscordService implements AutoCloseable
 	@Inject
 	private ScheduledExecutorService executorService;
 
-	private final DiscordEventHandlers discordEventHandlers = new DiscordEventHandlers();
-	private final DiscordRPC discordRPC = DiscordRPC.INSTANCE;
+	private DiscordRPC discordRPC;
 
 	/**
 	 * Initializes the Discord service, sets up the event handlers and starts worker thread that will poll discord
@@ -66,6 +65,18 @@ public class DiscordService implements AutoCloseable
 	public void init()
 	{
 		log.info("Initializing Discord RPC service.");
+
+		try
+		{
+			discordRPC = DiscordRPC.INSTANCE;
+		}
+		catch (UnsatisfiedLinkError e)
+		{
+			log.warn("Failed to load Discord library, Discord support will be disabled.");
+			return;
+		}
+
+		final DiscordEventHandlers discordEventHandlers = new DiscordEventHandlers();
 		discordEventHandlers.ready = this::ready;
 		discordEventHandlers.disconnected = this::disconnected;
 		discordEventHandlers.errored = this::errored;
@@ -83,7 +94,10 @@ public class DiscordService implements AutoCloseable
 	@Override
 	public void close()
 	{
-		discordRPC.Discord_Shutdown();
+		if (discordRPC != null)
+		{
+			discordRPC.Discord_Shutdown();
+		}
 	}
 
 	/**
@@ -95,6 +109,11 @@ public class DiscordService implements AutoCloseable
 	 */
 	public void updatePresence(DiscordPresence discordPresence)
 	{
+		if (discordRPC == null)
+		{
+			return;
+		}
+
 		final DiscordRichPresence discordRichPresence = new DiscordRichPresence();
 		discordRichPresence.state = discordPresence.getState();
 		discordRichPresence.details = discordPresence.getDetails();
@@ -127,7 +146,10 @@ public class DiscordService implements AutoCloseable
 	 */
 	public void clearPresence()
 	{
-		discordRPC.Discord_ClearPresence();
+		if (discordRPC != null)
+		{
+			discordRPC.Discord_ClearPresence();
+		}
 	}
 
 	/**
@@ -138,7 +160,10 @@ public class DiscordService implements AutoCloseable
 	 */
 	public void respondToRequest(String userId, DiscordReplyType reply)
 	{
-		discordRPC.Discord_Respond(userId, reply.ordinal());
+		if (discordRPC != null)
+		{
+			discordRPC.Discord_Respond(userId, reply.ordinal());
+		}
 	}
 
 	private void ready()

